ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml

```diff
@@ -29,154 +29,163 @@
+    state: present
+    src: "{{ playbook_dir }}/../../cluster-bootstrap/argocd-apps/{{ ENV }}/{{ item }}/application.yaml"
+  loop:
+    - "acm"
+    - "multicluster-observability"
+    - "grafana-dev"
+    - "prometheus-config"
+    - "patch-operator"
+    - "openshift-config"
+
+- name: Apply ArgoCD Applications for {{ ENV }} environment
+  kubernetes.core.k8s:
+    state: present
+    src: "{{ playbook_dir }}/../../cluster-bootstrap/argocd-apps/{{ ENV }}/{{ item }}/application.yaml"
+  loop:
+    - "group-sync"
+    - "sso"
+    - "alertmanager-to-github"
+  when:
+    ENV != 'local'
 
-    printf "=====================Create Alert manager config Argocd application ...\n"
-    if [ {{ ENV }} == "local" ]; then
-      sed -i'' --expression 's@<SLACK_API_URL>@'{{ SLACK_API_URL }}'@g' ./../../cluster-bootstrap/alert-manager-config/overlay/local/app-alertmanager-config.yaml
-      kubectl apply -k ./../../cluster-bootstrap/alert-manager-config/overlay/local
-    else
-      kubectl apply -k ./../../cluster-bootstrap/argocd-apps/{{ ENV }}/alert-manager-config
-    fi
-    printf "✓ Alert Manager configuration created!"
-  register: result
+- name: Wait for ACM Route creation
+  kubernetes.core.k8s_info:
+    kind: Route
+    name: multicloud-console
+    namespace: open-cluster-management
+  register: acm_route
+  until:
+    - acm_route.resources.0 is defined 
+    - acm_route.resources.0.status.ingress.0.conditions.0.type == 'Admitted'
+  retries: 15
+  delay: 20
+
+- name: Wait for MCO Route creation
+  kubernetes.core.k8s_info:
+    kind: Route
+    name: alertmanager
+    namespace: open-cluster-management-observability
+  register: mco_route
+  until:
+    - mco_route.resources.0 is defined
+    - mco_route.resources.0.status.ingress.0.conditions.0.type == 'Admitted'
+  retries: 15
+  delay: 20
+
+- set_fact:
+    acm_console_url: "{{ acm_route.resources.0.spec.host }}"
+    mco_console_url: "{{ mco_route.resources.0.spec.host }}"
+
+- name: Create a console-url secret
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      stringData:
+        ACM_CONSOLE_URL: "{{ acm_console_url }}"
+        ALERTMANAGER_URL: "{{ mco_console_url }}"
+      kind: Secret
+      metadata:
+        name: console-url
+        namespace: openshift-gitops
+      type: Opaque
+
+- name: Config Alert Manager for {{ ENV }} environment
+  kubernetes.core.k8s:
+    state: present
+    src: "{{ playbook_dir }}/../../cluster-bootstrap/argocd-apps/{{ ENV }}/{{ item }}/application.yaml"
+  loop:
+    - "alert-manager-config"
+  when:
+    ENV != 'local'
 
-- name: debug
-  debug:
-    msg: "{{ result.stdout_lines }}"
+- name: Config Alert Manager for local deployment
+  shell: |-
+    sed -i'' --expression 's@<SLACK_API_URL>@'{{ SLACK_API_URL }}'@g' {{ playbook_dir }}/../../cluster-bootstrap/alert-manager-config/overlay/local/app-alertmanager-config.yaml
+    kubectl apply -k ./../../cluster-bootstrap/alert-manager-config/overlay/local
```

Comment:
can also be replaced with https://docs.ansible.com/ansible/latest/collections/kubernetes/core/kustomize_lookup.html#ansible-collections-kubernetes-core-kustomize-lookup I think, as it is the last kubectl remaining part

Reply:
Update and abstract it to a template.